src/pip/_internal/commands/cache.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import absolute_import
+
+import logging
+import os
+import textwrap
+
+from pip._internal.cli.base_command import Command
+from pip._internal.exceptions import CommandError
+from pip._internal.utils.filesystem import find_files
+
+logger = logging.getLogger(__name__)
+
+
+class CacheCommand(Command):
+    """
+    Inspect and manage pip's caches.
+
+    Subcommands:
+        info:
+            Show information about the caches.
+        list [name]:
+            List filenames of packages stored in the cache.
+        remove <pattern>:
+            Remove one or more package from the cache.
+            `pattern` can be a glob expression or a package name.
+        purge:
+            Remove all items from the cache.
+    """
+    actions = ['info', 'list', 'remove', 'purge']
+    name = 'cache'
+    usage = """
+      %prog <command>"""
+    summary = "View and manage which packages are available in pip's caches."
+
+    def __init__(self, *args, **kw):
+        super(CacheCommand, self).__init__(*args, **kw)
+
+    def run(self, options, args):
+        if not args:
+            raise CommandError('Please provide a subcommand.')
+
+        if args[0] not in self.actions:
+            raise CommandError('Invalid subcommand: %s' % args[0])
+
+        self.wheel_dir = os.path.join(options.cache_dir, 'wheels')
+
+        method = getattr(self, 'action_%s' % args[0])
+        return method(options, args[1:])
+
+    def action_info(self, options, args):
+        format_args = (options.cache_dir, len(self.find_wheels('*.whl')))
+        result = textwrap.dedent(
+            """\
+            Cache info:
+              Location: %s
+              Packages: %s""" % format_args
+        )
+        logger.info(result)
+
+    def action_list(self, options, args):
+        if args and args[0]:
+            pattern = args[0]
+        else:
+            pattern = '*'
+
+        files = self.find_wheels(pattern)
+        wheels = map(self._wheel_info, files)
+        wheels = sorted(set(wheels))
+
+        if not wheels:
+            logger.info('Nothing is currently cached.')
+            return
+
+        result = 'Current cache contents:\n'
+        for wheel in wheels:
+            result += ' - %s\n' % wheel
+        logger.info(result.strip())
+
+    def action_remove(self, options, args):
+        if not args:
+            raise CommandError('Please provide a pattern')
+
+        files = self.find_wheels(args[0])
+        if not files:
+            raise CommandError('No matching packages')
+
+        wheels = map(self._wheel_info, files)
+        result = 'Removing cached wheels for:\n'
+        for wheel in wheels:
+            result += '- %s\n' % wheel
+
+        for filename in files:
+            os.unlink(filename)
+        logger.info(result.strip())
```

Comment:
yeah, agreed. seems like i kinda overengineered that bit.

Reply:
Switched to this. It now prints `Removed <number of files> files` by default, and if you've passed `--verbose` (or otherwise cause `logger.debug()` to print things) it should print `Removed <filename>` for each filename, as well.